### DISCUSSION TITLE
how to use arabert **V2** model using git clone https://github.com/

### DISCUSSION THREAD

**Discussion by salama4ai** [aub-mind/arabert#96] [Q&A]
i need to use arabert **V2** model using `git clone https://github.com/.....`

**Comment by WissamAntoun** [comment]
`git clone https://github.com/aub-mind/arabert` ? or do you mean you want the checkpoints?

**Reply by salama4ai** [reply]
isn't that for the old version? i was thought that it's for old repo :)
![Untitled](https://user-images.githubusercontent.com/76913472/115654652-da87f400-a331-11eb-9d37-7c6d1c0e1fff.png)

**Reply by WissamAntoun** [reply]
for the old repo, you need to also `git checkout` the commit hash

**Comment by mapmeld** [comment]
I'd also add, you can clone the model files down from the HuggingFace repo, but you need git-lfs support

```
curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
sudo apt-get install git-lfs
```

**Reply by salama4ai** [reply]
that's it

**Comment by salama4ai** [comment]
could i get the file -(.txt or .json orany format)- that used for pretraining your models (BERT,.....)?

**Reply by WissamAntoun** [reply]
I don't think we can. The files are >77GB and we can't re-distribute it. But most of it is already available for download for free.

**Reply by salama4ai** [reply]
to be more determinded, i just need to see few examples of the data that used for pretraining your models(BERT, GPT2,......), need to see few examples of this data after doing all preprocessing operations on it and just before feeding it to the model,
as examples:-
 is this data in shape of dialoge? 
A:....................................
B:....................................
A:...................................
B:....................................
OR are they have labels?
text:..................................
label:.............
or are they in shape of sentence and the wikipedia page that this sentence extracted from?
text:....................................
source of this text:.......................

i.e what was the objective while pretraining the models?? or what was the objective of model pretraining? is the goal is to train the model how to predict the next word of the sentence? or the goal is to predict the label of this sentence ? or ............

**Reply by WissamAntoun** [reply]
The data consists of documents with each sentence on a line, and documents separated by a new line.

AraBERT was trained using Masked language modeling and next sentence prediction, and AraGPT2 using next word prediction (or causal language modeling)

**Reply by salama4ai** [reply]
i can't import arabertpreprocessing
every time i run this:-

`pip install pytorch-transformers`
`from arabert.preprocess import ArabertPreprocessor`
`model_name = "aubmindlab/bert-base-arabertv2"`
`arabert_prep = ArabertPreprocessor(model_name=model_name)`
`text = "ولن نبالغ إذا قلنا: إن 'هاتف' أو 'كمبيوتر المكتب' في زمننا هذا ضروري"`
`arabert_prep.preprocess(text)`




![arabert1](https://user-images.githubusercontent.com/76913472/116419377-48a24e80-a83d-11eb-9c84-5b66ca0ce9a1.png)
i get error: no arabert module
![arabert](https://user-images.githubusercontent.com/76913472/116419395-4dff9900-a83d-11eb-8b04-8222817fe001.png)

**Reply by WissamAntoun** [reply]
you did clone Arabert repo: add `!git clone https://github.com/aub-mind/arabert`

**Reply by salama4ai** [reply]
when i clone arabert repo i get
![arabert2](https://user-images.githubusercontent.com/76913472/116420919-b1d69180-a83e-11eb-986f-b221fb3aea79.png)

**Reply by salama4ai** [reply]
the problem solved partially when i add `!pip install pyarabic`

but still not work yet, as i get :-
![arabert3](https://user-images.githubusercontent.com/76913472/116422617-370e7600-a840-11eb-824e-18359763b973.png)

**Reply by WissamAntoun** [reply]
!pip install farasapy

**Reply by salama4ai** [reply]
i think the problem is completely solved

the code i execute was
`!git clone https://github.com/aub-mind/arabert`
`!pip install pytorch-transformers`
`!pip install spacy ftfy==4.4.3`
`!pip install pytest`
`!pip install pyarabic`
`!pip install farasapy`
`# !python -m pytest -sv ./pytorch_transformers/tests/`
`# !python -m pytest -sv ./examples/`

`from arabert.preprocess import ArabertPreprocessor`
`# import Arabert`
`model_name = "aubmindlab/bert-base-arabertv2"`
`arabert_prep = ArabertPreprocessor(model_name=model_name)`

`text = "ولن نبالغ إذا قلنا: إن 'هاتف' أو 'كمبيوتر المكتب' في زمننا هذا ضروري"`
`arabert_prep.preprocess(text)`

**Reply by WissamAntoun** [reply]
please use `transformers` instead of `pytorch-transformers`

**Reply by salama4ai** [reply]
thank you soOoOoO Much i appreciate your efforts 👍

**Reply by salama4ai** [reply]
> 
> 
> please use `transformers` instead of `pytorch-transformers`

Ok i will

**Reply by salama4ai** [reply]
the final code :-
```
# !pip install spacy ftfy==4.4.3
# !pip install pytorch-transformers
# !pip install pytest
# !python -m pytest -sv ./pytorch_transformers/tests/
# !python -m pytest -sv ./examples/
```

`!git clone https://github.com/aub-mind/arabert`
`!pip install transformers`
`!pip install pyarabic`
`!pip install farasapy`
`from arabert.preprocess import ArabertPreprocessor`
`model_name = "aubmindlab/bert-base-arabertv2"`
`arabert_prep = ArabertPreprocessor(model_name=model_name)`

`text = "ولن نبالغ إذا قلنا: إن 'هاتف' أو 'كمبيوتر المكتب' في زمننا هذا ضروري"`
`arabert_prep.preprocess(text)`

**Reply by WissamAntoun** [reply]
yes looks good

**Reply by salama4ai** [reply]
by the way i repeat my question in other words to guarantee that the reader get what i mean 

By now a good job had done ,
now what i need to do:
i need to fine tune your model(arabert) using my data so i would apply your preprocessing method over my data through (
`model_name = "aubmindlab/bert-base-arabertv2"`
`arabert_prep = ArabertPreprocessor(model_name=model_name)`
`arabert_prep.preprocess(MY_DATA)`
)
after doing preprocessing, how can i fine tune your model using my data??
or how can i use your model as a start point to implement more advanced tasks"question-answering bot" as example??
you pretrain BERT model to produce preptrained bert model, the next step is to use my data to fine-tune your model to be able to make more advanced tasks such as (Question-Answering bot or chitchat bot or .......) how can i do that? 
if i get your model weights so i will be able to fine-tune your model, in other words assume i get more data and i need to pretrain your arabert model further from last point you pretrain your model ,how can i do that???????????????

**Reply by WissamAntoun** [reply]
This is a big question.  you can build a question-answering bot that can extract answers for documents using AraBERT and the TyDIQA dataset. Then you need to build a system that finds relevant documents so BERT can search through them. An example of this is available in my other repository https://github.com/WissamAntoun/arabic-wikipedia-qa-streamlit which you can try here https://share.streamlit.io/wissamantoun/arabic-wikipedia-qa-streamlit/main

**Reply by salama4ai** [reply]
ok, so i stuck now in finding your model weights, i mean i need your bert model as if i will continue pretrain arabert with additional data, that's it,

**Reply by WissamAntoun** [reply]
Arabert v0.1 and v1 tensorflow models are available here https://drive.google.com/drive/folders/1xqNwdlaMXabFzCUtmVwQZDPbCib4U6zn?usp=sharing 

all other models (pytorch, tensorflow 1 and 2) are available on huggingface's library https://huggingface.co/aubmindlab

**Reply by salama4ai** [reply]
thanks alot , that's what i were searching 👍

**Reply by BassamArkok** [reply]
I installed farasapy but I get this message when I run this command "arabert_prep = ArabertPreprocessor(model_name=model_name)"
WARNING:root:farasapy is not installed, you want be able to process text for AraBERTv1 and v2. Install it using: pip install farasapy.

**Reply by BassamArkok** [reply]
please how is to fix this problem?
also, I get this error message "Traceback (most recent call last):
  File "<pyshell#9>", line 1, in <module>
    arabert_prep.preprocess(text)
  File "C:\Python\Python395\arabert\preprocess.py", line 257, in preprocess
    text = self.farasa_segmenter.segment(text)
AttributeError: 'ArabertPreprocessor' object has no attribute 'farasa_segmenter'"
after the command "arabert_prep.preprocess(text)"

**Reply by salama4ai** [reply]
try doing 
`pip install farasapy`

**Reply by BassamArkok** [reply]
I already installed farasapy but the problem is still
